crates/resolver/src/config.rs

```diff
@@ -393,6 +393,15 @@ impl fmt::Display for NameServerConfig {
     }
 }
 
+/// We consider a `NameServerConfig` equal if the `socket_addr` and `protocol` are equal
+impl PartialEq for NameServerConfig {
+    fn eq(&self, other: &Self) -> bool {
+        self.socket_addr == other.socket_addr && self.protocol == other.protocol
```

Comment:
I'm not sure if this is a complete enough equality statement. Can you explain a little more? That is, all the other fields would be important differences that would make a NameServerConfig either work or not...

Reply:
Basically, `rustls::ClientConfig` can't actually be compared and I don't think we should paper over that by wrapping it and faking equality for it based on `Arc` pointer. It seems to me that the `socket_addr` + `protocol` is sort of the "identity" of the `NameServerConfig` and everything else is config necessary to make the connections work, so this seems reasonable to me. The main usage outside tests seem to be in reordering name servers for the `ServerOrderingStrategy::QueryStatistics`, for which I don't think this really needs to be complete.

Reply:
Ok, I just worry that it might have awkward behavior in cases where people might want a deep equality check.

Reply:
Follow-up in https://github.com/hickory-dns/hickory-dns/pull/2579.